lib/discovery.js

```diff
@@ -0,0 +1,198 @@
+import { TypedEmitter } from "tiny-typed-emitter";
+import Hyperswarm from "hyperswarm";
+
+import { LocalPeers } from "./localpeers.js";
+
+/**
+ * @typedef {Object} DiscoveryEvents
+ * @property {(connection, info) => void} peer
+ * @property {(error: Error) => void} error
+ */
+
+/**
+ * @extends {TypedEmitter<DiscoveryEvents>}
+ */
+export class Discovery extends TypedEmitter {
+  #peers = new Map();
+  #discover = new Map();
+  #dht;
+  #mdns;
+
+  /**
+   * @param {object} options
+   * @param {number} options.port
```

Comment:
This is setting the port for the mdns discovery, but I hyperswarm is choosing its own port ([49737 by default](https://github.com/hyperswarm/dht/blob/master/index.js#L22)). I wonder if we just let the localpeers class handle the port? Do we risk just choosing a very random port (like hyperswarm) and assuming no other service will be on it? One thing to bear in mind is different instances of Mapeo running on the same device (QA build and production) which can cause issues with fixed port numbers.

Reply:
I realized we can reasonably let the net `server.listen` handle picking an available port.